src/clean_air/data/data_store_access.ipynb

```diff
@@ -0,0 +1,354 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Accessing the Clean Air Framework Datastore\n",
+    "Data is stored on an object store on the JASMIN cloud as a \"dataset\".\n",
+    "A dataset is composed of all the datafiles that hold the data (e.g. netcdf or CSV files) plus metadata about the dataset.\n",
+    "Metadata can be accessed independently of the data, so that you don't have to download the entire dataset if you don't need it.\n",
+    "\n",
+    "Data objects are provided to represent the stored data.\n",
+    "Datasets are encapsulated by the `DataSet` class, whilst metadata is encapsulated by the `Metadata` class.\n",
+    "Access to stored datasets and metadata are via the `S3FSDataSetStore` and `S3FSMetdataStore` classes respectively.\n",
+    "(`s3fs` is the underlying library used to access the object store)\n",
+    "\n",
+    "Both data access classes provide a similar interface, working with their respective data objects."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Credentials for Object Store Access\n",
+    "Anonymous access is enabled for the data bucket, but to be able to upload data credentials with write permissions are \n",
+    "required.\n",
+    "\n",
+    "Documentation on `s3fs`'s credentials are here: https://s3fs.readthedocs.io/en/latest/#credentials\n",
+    "As it uses Amazon AWS's `botocore` library, the credential config is the same as for `botocore` & `boto3`.\n",
+    "\n",
+    "In summary, the options are (in order of increasing precedence):\n",
+    "\n",
+    "* `~/.aws/credentials` file (refer to [Configuration and credential file settings](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)) \n",
+    "with contents such as:\n",
+    "```\n",
+    "[default]\n",
+    "aws_access_key_id=AKIAIOSFODNN7EXAMPLE\n",
+    "aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY\n",
+    "```\n",
+    "* Setting the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variables\n",
+    "* Explicitly creating an `s3fs.S3FileSystem` instance and passing them as arguments (refer to next section for details)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Creating a DataSetStore or MetadataStore\n",
+    "### Helper Functions\n",
+    "Helper functions are available that create instances of both `S3FSDataSetStore` and `S3FSMetadataStore`:\n",
+    "* `create_dataset_store`\n",
+    "* `create_metadata_store`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
```

Comment:
Jupyter notebooks are horrible to store in version control. The best practice I've seen in other projects is to always store unexecuted notebooks (that is, where all code cell output has been cleared). That way the only changes that should be in modified notebooks are actual changes to the notebook, not inline metadata like this. Of course different might be being done in this project - this is just a thought!

Reply:
In this case, the notebook is acting as documentation, so having the result is desirable. Although I would prefer it if the execution count was always `1`

If we do want to strip out this stuff though, this looks like a handy way of automating it: https://zhauniarovich.com/post/2020/2020-10-clearing-jupyter-output-p3/